aperturedb/ParallelQuery.py

```diff
@@ -316,7 +333,8 @@ def worker(self, thid: int, generator, start: int, end: int):
             batch_end = min(batch_start + self.batchsize, end)
 
             try:
-                self.do_batch(db, generator[batch_start:batch_end])
+                self.do_batch(db, batch_start,
+                              generator[batch_start:batch_end])
```

Comment:
It's probably a bad idea to call this parameter "generator" if we're going to index it like this.

Reply:
I assume you are talking in terms of a python generator - my assumption was that it was named this because it is a QueryGenerator ( or was initially )